Jenkinsfile

```diff
@@ -332,6 +348,35 @@ pipeline {
                 checkout scm
             }
         }
+        stage('check_for_changed_files') {
+          steps {
+            script {
+	      if (env.BRANCH_NAME == 'master') {
+	        return
+	      }
+
+	      def all_files = new HashSet<String>()
+
+              for (set in currentBuild.changeSets) {
+                def entries = set.items
+                for (entry in entries) {
+                  for (file in entry.affectedFiles) {
+                    all_files << file.path
+                  }
+                }
+              }
+
+	      echo "All files changed since last build: ${all_files}"
```

Comment:
Nit: indent is off, it seems.  It's also off in the `if` block below.

Reply:
leading tab instead spaces